src/Solver/Numerov.cpp

```diff
@@ -0,0 +1,249 @@
+        }
+    }
+
+}
+
+
+std::vector<double> Numerov::findProbability() {
+    std::vector<double> probability = this->tempWavefunction; 
+    std::transform(probability.begin(), probability.end(), probability.begin(), square);
+    return probability;
+}
+
+void Numerov::normalize(){
+    double norm = this->trapezoidalRule(0.0, this->nbox, dx, this->probability);
+    std::transform(this->tempWavefunction.begin(), this->tempWavefunction.end(), this->tempWavefunction.begin(), [norm](double value) { return value / sqrt(norm); });
+}
+
+double Numerov::findEnergy(double e_min, double e_max, double e_step, std::vector<double> &this_potential) {
+    double solutionEnergy = 0;
+    int sign;
+    // scan energies to find when the Numerov solution is = 0 at the right extreme of the box.
+    for (int n = 0; n < (e_max - e_min) / e_step; n++)
+    {
+        double energy = e_min + n * e_step;
+        this->findWavefunction(energy, this_potential);
+        double &last_wavefunction_value = this->tempWavefunction[this->nbox];
+
+        if (fabs(last_wavefunction_value) < err)
+        {
+            std::cout << "Solution found" << last_wavefunction_value << std::endl;
+            solutionEnergy = energy;
+            break;
+        }
+
+        if (n == 0)
+            sign = (last_wavefunction_value > 0) ? 1 : -1;
+
+        // when the sign changes, means that the solution for f[nbox]=0 is in in the middle, thus calls bisection rule.
+        if (sign * last_wavefunction_value < 0)
+        {
+            std::cout << "Bisection " << last_wavefunction_value << std::endl;
+            solutionEnergy = this->bisection(energy - e_step, energy + e_step, this_potential);
+            break;
+        }
+    }
+    return solutionEnergy;
+}
+
+std::ostream& operator<<(std::ostream& stream, Numerov& solver) {
+
+	// Print solution energy
+    stream << "Solution energy: " << solver.getSolutionEnergy() << std::endl;
+    
+    // Print wavefunction
+    if (solver.getWavefunction().size() > 0) {
+        stream << "Wavefunction: " << std::endl;
+        for (double x : solver.getWavefunction())
+            stream << "\t" << x << std::endl;
+    } else 
+        stream << "Wavefunction not evaluated." << std::endl;
+
+	// Print probability
+    if (solver.getProbability().size() > 0) {
+        stream << "Probability: " << std::endl;
+        for (double x : solver.getProbability())
+            stream << "\t" << x << std::endl;
+    } else 
+        stream << "Probability not evaluated." << std::endl;
+
+    return stream;
+ }
+
+// If we have separable potentials V(a, ... ,z) = V(a)+ ... +V(z), then we have a resulting wavefunction
+// that in the form W(a,...,z) = W(a) * ... * W(z). With this routine, each time a new wavefunction
+// is evaluated, update the final wavefunction multiplying it with the new one.
+// So, it takes a W(x) wavefunction as a parameter and multiply it with the global W(a, ... ,z)
+void Numerov::multiplyWavefunction() {
```

Comment:
Doesn't this multiply element by element?

We need `// W(x,y) = W(x) * W(y)`, that means 
```
for each i, j: 
     wf[i, j] = wf1[i]*wf2[j]
```

Suggestion: use overload of `^` ? @mechanapoleon

Reply:
That's what it does.
The final wavefunction is `this->wavefunction` (aka W(x,y) ), and for each W(k) where k \in {x, y}, in the code called `this->tempWavefunction` we multiply it to the final one using a `std::transform`